django/db/models/options.py

```diff
@@ -295,8 +298,22 @@ def _prepare(self, model):
         else:
             self.order_with_respect_to = None
 
+        if self.primary_key is not None and (
+            not isinstance(self.primary_key, (tuple, list))
+            or not self.primary_key
+            or not all(isinstance(field, str) for field in self.primary_key)
+            or len(set(self.primary_key)) != len(self.primary_key)
+        ):
+            raise ValueError(
+                "The value of 'Meta.primary_key' must be a non-empty tuple or list "
+                "of unique strings."
+            )
+
         if self.pk is None:
-            if self.parents:
+            if self.primary_key:
+                pk = CompositePrimaryKey(*self.primary_key)
+                model.add_to_class("primary_key", pk)
```

Comment:
Looking through the PR I noticed that `primary_key` doesn't seem to be used anywhere as far as I can tell. There are a few test checking that it is equal with `pk` but that seems to be about it? Any thoughts on that @csirmazbendeguz

Reply:
This is `model.primary_key` (not `model._meta.primary_key`) if I understand correctly.

Reply:
The idea is if Meta.primary_key is defined it adds a field called primary_key to the model. This is the composite field CompositePrimaryKey. This field is not supposed to be accessed directly, users should access pk instead. If it could be hidden from users it would be best.

Reply:
If it is not supposed to get accessed directly, why do we need it all -- ie which purpose does it serve?

Reply:
@apollo13 , a virtual, composite field is the simplest way to implement this. The question is rather if we need the `Meta.primary_key` option or not. The people I talked to so far didn't have anything against it, I think it's a nice API.

Reply:
We seem to be talking about different things so I will rephrase the question:
What for do we need `Model.primery_key` (not `Meta.primary_key`) for? The tests don't show it used anywhere and the docs don't mention it either making me think that it might not be required at all?

Reply:
I think the question is "Why do we have both `pk` and `primary_key`? Could we have just `pk` instead?".

Reply:
The sole reason we need it is to point `pk` _somewhere_

Reply:
Ah, thank you for clarifying. I would recommend to name it `_primary_key` then or has this any downsides?

Reply:
Yeah we could, the name of this field doesn't matter since reference to it is retained via `pk`.

This line is actually the only questionable thing about the `Meta.primary_key` approach, we still have to add a field.

Reply:
Actually, we could also add a random suffix to the end, just to make sure people don't access it.

Reply:
There is no need for randomness here. Every undocumented attribute is private as far as Django is concerned. If people use it they are on their own.

Reply:
Alright